libnopegl/src/node_transform.c

```diff
@@ -68,8 +69,19 @@ static int transform_init(struct ngl_node *node)
 
 static int transform_update(struct ngl_node *node, double t)
 {
-    const struct transform_opts *o = node->opts;
-    return ngli_node_update(o->child, t);
+    struct transform_priv *s = node->priv_data;
+    struct transform_opts *o = node->opts;
+
+    int ret = ngli_node_update_children(node, t);
+    if (ret < 0)
+        return ret;
+
+    if (o->matrix_node) {
+        float *data = ngli_node_get_data_ptr(o->matrix_node, o->matrix);
+        memcpy(s->trf.matrix, data, sizeof(s->trf.matrix));
+    }
```

Comment:
Don't you need to update the matrix first, then the update of the child?

Reply:
In the update phase, the children must be updated first so the parents can rely on them.